### ISSUE TITLE
Text `letterSpacing > 0` cause the RTL text to flip

### ISSUE THREAD

**Issue by nisimjoseph** [pixijs/pixijs#7729]
### Expected Behavior
RTL text behaves the same way with `letterSpacing` is 0 or above.

### Current Behavior
the text flip if `letterSpacing`  > 0

### Possible Solution
give another param to control if the RTL characters are flip or not.

### Steps to Reproduce
see this example link:
https://pixijs.io/pixi-text-style/#%7B%22style%22%3A%7B%22align%22%3A%22right%22%2C%22fontFamily%22%3A%22Alef-regular%22%2C%22fontSize%22%3A33%2C%22letterSpacing%22%3A1%2C%22trim%22%3Atrue%2C%22whiteSpace%22%3A%22normal%22%7D%2C%22text%22%3A%22.%D7%AA%D7%99%D7%A8%D7%91%D7%A2%20%D7%93%D7%95%D7%A2%D7%95%2010%20%D7%AA%D7%99%D7%A8%D7%91%D7%A2%20%D7%AA%D7%A7%D7%99%D7%93%D7%91%5Cntest%20text%201%22%7D

just play with the `Letter Spacing` value to 0 (zero) and 1 to see the issue.

### Environment
version 6.1.1

**Comment by bigtimebuddy** [comment]
Support for RTL is definitely a feature request. PR welcome for someone that wants to add!

**Comment by nisimjoseph** [comment]
thank you. I have about it. I research today about the RTL and notice the browser renders RTL with LTR text inside canvas just like it does it in non-canvas. but I used `canvas.getContext("2d")` to do it and not `webgl2`/`webgl` type.
I am trying to mix `webgl` and `2d` types or render text using `webgl` only but I got to missing Text rendering function (`ctx.fillText`).
can we mix those 2 Contexts somehow? 
I will keep digging on it, but thought maybe someone has a lead about it.

**Comment by bigtimebuddy** [comment]
The Text object in Pixi uses a context2d to render text. The issue here is that when letter-spacing is applied (> 0) we manually draw each character. There's an assumption of LTR. The layout needs to, ideally, support RTL depending on the text.

**Comment by eXponenta** [comment]
Another unresolved problem:
A lot of RTL Language cannot use a letter spacing, because they use a ligatures.
If you will add spacing - ligatures will be broken.

**Comment by bigtimebuddy** [comment]
That's true @eXponenta so this is probably be a much bigger task for ligature-heavy languages

**Comment by nisimjoseph** [comment]
Hi,
Want to share a thought about Text rendering which might make it easier.
How about creating another Canvas that has 2D Context, the canvas is not in the DOM, and we draw the text on it and copy the Text's content as PNG with alpha. 
Each change in the text will generate a new PNG for this text with parameters for size, font, and others.
The canvas drawing any text with perfect text handling, which you note above, might be a solution until real text rendering is available.

Will be happy for your thoughts about it.

**Comment by eXponenta** [comment]
@nisimjoseph Hmmm, good idea!

How you think what pixi doing now?

Plz, checking a source before submitting any such ideas. 
https://github.com/pixijs/pixijs/blob/dev/packages/text/src/Text.ts#L177
https://github.com/pixijs/pixijs/blob/dev/packages/text/src/Text.ts#L352
This nothing will solve when you will add custom letters spacing, and a converting a canvas to PNG is overhead because canvas can be uploaded directly.

**Comment by eXponenta** [comment]
@bigtimebuddy How you think, can we add modern canvas API for text with fallback onto current implementation?

https://github.com/fserb/canvas2D/blob/master/spec/text-modifiers.md

Origin Trial for Chrome already can activate it (available from 92) and API will arrived in 95 Chrome by default.

https://developer.chrome.com/origintrials/#/view_trial/3585991203293757441

**Comment by ShukantPal** [comment]
@bigtimebuddy @eXponenta Should this be closed given the PR?

**Comment by Zyie** [comment]
Hi there! 👋

We're doing some housekeeping and closing out older issues that haven't had recent activity.

PixiJS has evolved significantly, and we only supporting v7 and above. If this issue is still affecting you on PixiJS v8 (latest), please feel free to create a new issue with updated details and a minimal repro case.

Thanks for your understanding and for helping us keep the issue tracker focused and maintainable! 🩷
